tests/test_tasker.py

```diff
@@ -369,3 +374,33 @@ def version(self):
         .once())
 
     DockerTasker()
+
+
+def my_func(*args, **kwargs):
+    my_args = ('some', 'new')
+    my_kwargs = {'one': 'first', 'two': 'second'}
+    assert args == my_args
+    assert kwargs == my_kwargs
+    response = requests.Response()
+    response.status_code = 408
+    raise APIError("test fail", response)
+
+
+@pytest.mark.parametrize('retry_times', [0, 1, 2, 3])
+def test_retry_method(retry_times):
```

Comment:
This test seems to verify that function is not being called when `retry_times == 0` and being called at least once in other cases. 

It should also verify that the wrapped function is being called several times, should check that timeout is calculated correctly, should check that the exception is propagated after a number of attempts etc.

Reply:
it is checking how many times wrapped function, so you mean it should check also the wrapper function? well if wrapped function is called 3 times, wrapper has to be called as well

Reply:
It should at least check that the exception is thrown after a number of retries

Reply:
mmm? there is pytest.raises which is expecting APIError if retry_times is >0

Reply:
Right, but does it check that exception is thrown after a specific number of attempts? It doesn't, but it should be, in order to verify that the request is being retried more than once.

Reply:
this is checking that function is called X times, and after that whole retry will raise

 (flexmock(sys.modules[__name__])
        .should_call('my_func')
        .with_args(*my_args, **my_kwargs)
        .times(retry_times)